src/metabase/driver/generic_sql/query_processor.clj

```diff
@@ -81,6 +88,17 @@
   (formatted [{unit :unit, field :field}]
     (sql/date (driver) unit (formatted field)))
 
+  BinnedField
+  (formatted [{{:keys [min-value max-value] :as field} :field bins :num-bins}]
+    (let [bin-width (Math/ceil (/ (- (Math/floor max-value)
```

Comment:
BTW, putting this logic in the protocol extensions was a nice refactor @tlrobinson. I stole it from an earlier version of https://github.com/metabase/metabase/pull/4700